smaug/python/ops/common.py

```diff
@@ -29,6 +29,11 @@ def check_and_add_layout_transform(name, op, input_tensors):
         op].input_layoutsets[i]
     input_layout = input_tensors[i].shape.layout
     if not expected_layoutset.contains(input_layout):
+      reorder_op_output = tensor_utils.get_tensor_reorder_op(
```

Comment:
This is such a brilliantly simple idea I can't believe we didn't think about this earlier :)